File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int[] cloneArr = arr.clone();
        Arrays.sort(cloneArr);
        int min = cloneArr[0];
        
        for(int i=0;i<arr.length;i++){
            if(arr[i] == min){
                for(int j=i;j<arr.length-1;j++){
                    int tmp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = tmp;
                }
                break;
            }
        }
        
        int[] answer = new int[arr.length - 1 > 0 ? arr.length-1 : 1];
        if(arr.length > 1){
            System.arraycopy(arr, 0, answer, 0, arr.length-1);
        } else
            answer[0] = -1;
        
        return answer;
    }
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.kt
class Solution {
    fun solution(arr: IntArray): IntArray {
        var answer = intArrayOf()
        
        if(arr.size == 1){
            return intArrayOf(-1)
        }
        
        answer = arr.filter { it != arr.minOrNull() }.toIntArray()
        
        return answer
    }
}